## 00 |

The objective of this **project** is to define the profile of the most profitable customers, based on a collection of **9.134** customer data, including demographics and buying behaviour.

In [5]:
# 📚 Basic libraries
import os # file managment
import pandas as pd # data manipulation
import numpy as np # numerical operations
import math # math in-build
import matplotlib.pyplot as plt # 2D visualization
import seaborn as sns # HD visualization
import warnings # warning messages managment

# 🤖 Machine Learning
from sklearn.preprocessing import StandardScaler # data normalization
from sklearn.model_selection import train_test_split # splitting data into train/test sets
from sklearn.linear_model import LinearRegression # logistic model
from sklearn.linear_model import ElasticNet, Lasso, Ridge # Regression model
from sklearn.metrics import r2_score, mean_squared_error, mean_absolute_error # model evaluation metrics

# ⚙️ Settings
pd.set_option('display.max_columns', None)
warnings.filterwarnings('ignore') # ignore warnings

# 🔄 Functions
import sys # system path to our functions
sys.path.append("../../IronData/01. GitHub/01. IronLabs/unit_4_py/lab-cleaning-numerical-data")

from easy_functions.functions import open_data # quick data overview
from easy_functions.functions import snake_columns # snake_case
from easy_functions.functions import explore_data # checks for duplicates, NaN & empty spaces

ModuleNotFoundError: No module named 'easy_functions'

## 01 | Data Extraction

In [2]:
file_path = os.path.join("C:/Users/apisi/01. IronData/01. GitHub/01. IronLabs/unit_4_py/lab-cleaning-numerical-data/01_data/we_fn_use_c_marketing_customer_value_analysis.csv")
data = pd.read_csv(file_path)
open_data(data)

Data shape is (9134, 24).

Customer                          object
State                             object
Customer Lifetime Value          float64
Response                          object
Coverage                          object
Education                         object
Effective To Date                 object
EmploymentStatus                  object
Gender                            object
Income                             int64
Location Code                     object
Marital Status                    object
Monthly Premium Auto               int64
Months Since Last Claim            int64
Months Since Policy Inception      int64
Number of Open Complaints          int64
Number of Policies                 int64
Policy Type                       object
Policy                            object
Renew Offer Type                  object
Sales Channel                     object
Total Claim Amount               float64
Vehicle Class                     object
Vehicle Size                  

Customer       State  Customer Lifetime Value Response Coverage  \
2328  AM26717  California              3123.080994      Yes    Basic   
4586  KL39986     Arizona             19278.817150       No    Basic   
2224  DQ50538      Nevada              9050.756606       No    Basic   
3540  BY23642     Arizona              4595.043966       No    Basic   
4572  KI70852      Nevada              3569.864625       No    Basic   

                 Education Effective To Date EmploymentStatus Gender  Income  \
2328              Bachelor           2/19/11       Unemployed      M       0   
4586  High School or Below            1/5/11       Unemployed      F       0   
2224              Bachelor           2/11/11         Employed      M   63800   
3540              Bachelor           1/17/11       Unemployed      F       0   
4572              Bachelor           2/27/11       Unemployed      F       0   

     Location Code Marital Status  Monthly Premium Auto  \
2328      Suburban        Married                   101   
4586      Suburban         Single                    61   
2224      Suburban        Married                   113   
3540      Suburban         Single                    64   
4572      Suburban         Single                   101   

      Months Since Last Claim  Months Since Policy Inception  \
2328                        1                             83   
4586                       12                             77   
2224                        9                             43   
3540                       16                             62   
4572                       12                             38   

      Number of Open Complaints  Number of Policies    Policy Type  \
2328                          4                   1  Personal Auto   
4586                          0                   2  Personal Auto   
2224                          0                   4  Personal Auto   
3540                          0                   6  Personal Auto   
4572                          0                   1  Personal Auto   

           Policy Renew Offer Type Sales Channel  Total Claim Amount  \
2328  Personal L1           Offer1         Agent          960.115399   
4586  Personal L3           Offer3        Branch          366.604732   
2224  Personal L3           Offer1        Branch          542.400000   
3540  Personal L3           Offer1        Branch          460.800000   
4572  Personal L3           Offer1   Call Center         1201.085226   

     Vehicle Class Vehicle Size  
2328           SUV        Large  
4586  Two-Door Car      Medsize  
2224           SUV      Medsize  
3540  Two-Door Car        Small  
4572           SUV        Small

💻 This **dataset** contains **9134 unique customer data** from **24 columns**.

We'll analyze demographics such as **Income**, **State**, **Education**, and **EmploymentStatus** and to understand their buying behavior, we'll examine features like **Vehicle Class**, **Monthly Premium Auto**, and **Policy Type**. Our target variable is the **Total Claim Amount**.

In [3]:
data_c = data.copy() # copy before doing any changeNdata
data_c.to_csv('C:/Users/apisi/01. IronData/01. GitHub/01. IronLabs/unit_4_py/lab-cleaning-numerical-data/01_data/data_c.csv')